src/logger.cc

```diff
@@ -166,10 +174,10 @@ NAN_METHOD(Logger::Critical) {
   }
 
   Logger *obj = Nan::ObjectWrap::Unwrap<Logger>(info.This());
-  const std::string message = *Nan::Utf8String(info[0]);
 
   if (obj->logger_) {
-    obj->logger_->critical(message);
+    const Nan::Utf8String message(info[0]);
+    obj->logger_->critical(utf8StringAsStringView(message));
```

Comment:
Why change to a string view here ? Is there an advantage with this version of the call ?

Reply:
Converting from Utf8String to std::string will lead to buffer copying. By using string view we can avoid the unnecessary copy.

Reply:
I believe the log calls are async, will the stringview be valid till the `logger_` functions finish ?

Reply:
You're correct. I will change them back.